protocol/testing/containertest/testnet_test.go

```diff
@@ -262,3 +275,60 @@ func TestMarketPrices(t *testing.T) {
 	}
 	assertPricesWithTimeout(t, node, expectedPrices, 30*time.Second)
 }
+
+func TestUpgrade(t *testing.T) {
+	testnet, err := NewTestnetWithPreupgradeGenesis()
+	require.NoError(t, err, "failed to create testnet - is docker daemon running?")
+	err = testnet.Start()
+	require.NoError(t, err)
+	defer testnet.MustCleanUp()
+	node := testnet.Nodes["alice"]
+
+	proposal, err := gov.NewMsgSubmitProposal(
+		[]sdk.Msg{
+			&upgrade.MsgSoftwareUpgrade{
+				Authority: govModuleAddress,
+				Plan: upgrade.Plan{
+					Name:   UpgradeToVersion,
+					Height: 10,
+				},
+			},
+		},
+		testapp.TestDeposit,
+		constants.AliceAccAddress.String(),
+		testapp.TestMetadata,
+		testapp.TestTitle,
+		testapp.TestSummary,
+	)
+	require.NoError(t, err)
+
+	require.NoError(t, BroadcastTx(
+		node,
+		proposal,
+		constants.AliceAccAddress.String(),
+	))
+	err = node.Wait(2)
+	require.NoError(t, err)
+
+	addresses := []string{
+		constants.AliceAccAddress.String(),
+		constants.BobAccAddress.String(),
+		constants.CarlAccAddress.String(),
+		constants.DaveAccAddress.String(),
+	}
+
+	for _, address := range addresses {
+		require.NoError(t, BroadcastTx(
```

Comment:
My understanding is this synchronous or asynchronous broadcast? What would cause `BroadcastTx` to return an error?

Reply:
It'll default to sync here. I'm bootstrapping by reusing some broadcast functions and those can error in various places before and during the actual broadcast. Realistically this will probably never error in practice though.